tests/macros/addonsidebar.test.js

```diff
@@ -0,0 +1,168 @@
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/author`,
+                title: 'author'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/background`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/background`,
+                title: 'background'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/theme`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/theme`,
+                title: 'theme'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/version`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/version`,
+                title: 'version'
+            },
+        ],
+        slug: MANIFEST_SLUG,
+        title: 'manifest.json'
+    };
+}
+
+function checkSidebarDom(html, locale, is_under_web_ext_api=false) {
+    // Lint the HTML
+    expect(lintHTML(html)).toBeFalsy();
+    const dom = JSDOM.fragment(html);
+    const section = dom.querySelector('section.Quick_links');
+    // Check the basics
+    expect(section).toBeTruthy();
+    // Check the total number of top-level list items that can be toggled
+    expect(
+        section.querySelectorAll('ol > li.toggle').length
+    ).toEqual(SUMMARIES[locale].size);
+    // Check that all links reference the proper locale or use https
+    const num_total_links = section.querySelectorAll('a[href]').length;
+    const num_valid_links = section.querySelectorAll(
+        `a[href^="/${locale}/docs/Mozilla/Add-ons"], a[href^="https://"]`
+    ).length;
+    expect(num_valid_links).toEqual(num_total_links);
+    // Check that one of the list item's details are open by default (or not)
+    const details_open = section.querySelector(
+        'ol > li.toggle > details[open]'
+    );
+    if (is_under_web_ext_api) {
+        expect(details_open).toBeTruthy();
+    } else {
+        expect(details_open).toBeFalsy();
+    }
+    // Check a sample of the DOM for localized content
+    for (const node of section.querySelectorAll('summary')) {
+        expect(SUMMARIES[locale].has(node.textContent)).toBe(true);
+    }
+    // Check for "WebExtensions/manifest.json" details
+    for (const name of ['author', 'background', 'theme', 'version']) {
+        const href = `/${locale}/docs/${MANIFEST_SLUG}/${name}`;
+        expect(section.querySelector(`li > a[href="${href}"]`)).toBeTruthy();
+    }
+}
+
+describeMacro('AddonSidebar', function() {
+    beforeEachMacro(function(macro) {
+        // Mock calls to template('WebExtAPISidebar', [])
+        macro.ctx.template = jest.fn((macro, args) => {
+            // This template will be tested on its own, so nothing needed here.
+            return '';
+        });
+        // Mock calls to MDN.fetchJSONResource(doc_url)
```

Comment:
It wasn't obvious to me that (from the point of view of AddonSidebar) this is mocking `tree` (https://github.com/mdn/kumascript/blob/master/macros/AddonSidebar.ejs#L390). I think it would be helpful to mention that here.